web/src/beta/features/Editor/Settings/index.tsx

```diff
@@ -27,22 +30,49 @@ const Settings: React.FC<Props> = ({
   currentCamera,
   layers,
   tab,
+  selectedPage,
   onFlyTo,
+  onPageUpdate,
 }) => {
   const t = useT();
-  const [layerCheck, setLayerCheck] = useState(true);
+  const [layerChecked, setLayerChecked] = useState<string[]>([]);
+
+  useEffect(() => {
+    if (selectedPage && layers) {
+      const selectedLayerIds = selectedPage.layersIds || [];
+      setLayerChecked(selectedLayerIds);
+    }
+  }, [selectedPage, layers]);
+
+  const pageId = selectedPage?.id;
+  const handleLayerCheck = (layerId: string) => {
+    setLayerChecked(prev => {
+      const updatedLayers = prev.includes(layerId)
+        ? prev.filter(id => id !== layerId)
+        : [...prev, layerId];
+
+      pageId && onPageUpdate?.(pageId, updatedLayers);
```

Comment:
To avoid the UI becoming incorrect we should:
- `if (!pageId) return;` at the top of handleLayerCheck
- return prev if onPageUpdate fails <- this is maybe ignorable if the performance becomes really bad... in the future I want to implement optimistic mutations, but until then the feedback of user's click might be too slow....